client/src/main/kotlin/io/spine/examples/shareaware/client/Client.kt

```diff
@@ -0,0 +1,237 @@
+        clientRequest()
+            .command(message)
+            .postAndForget()
+    }
+
+    /**
+     * Subscribes to the changes in entity of the given type.
+     *
+     * @param type type of the entity on which changes subscription works
+     * @param id entity ID by which arrived entities will be filtered
+     * @param observer callback function that will be triggered when the entity state changes
+     */
+    public fun <S : EntityState> subscribeToEntity(type: Class<S>, id: Message, observer: (S) -> Unit) {
+        clientRequest()
+            .subscribeTo(type)
+            .byId(id)
+            .observe(observer)
+            .post()
+    }
+
+    /**
+     * Subscribes to the events of the provided type.
+     *
+     * @param type type of the event to subscribe on
+     * @param filter filter by which arrived events of the provided type will be filtered
+     * @param observer callback function that will be triggered when the event arrives
+     */
+    public fun <E : EventMessage> subscribeToEvent(
+        type: Class<E>,
+        filter: EventFilter,
+        observer: (E) -> Unit
+    ) {
+        clientRequest()
+            .subscribeToEvent(type)
+            .where(filter)
+            .observe(observer)
+            .post()
+    }
+
+    /**
+     * Retrieves the entity with a provided type and ID.
+     *
+     * @param type type of the entity to be retrieved
+     * @param id entity ID by which the query result will be filtered
+     */
+    public fun <E : EntityState> readEntity(type: Class<E>, id: Message): E? {
+        val entities = clientRequest()
+            .select(type)
+            .byId(id)
+            .run()
+        if (entities.isEmpty()) {
+            return null
+        }
+        return entities[0]
+    }
+
+    /**
+     * Returns the ID of the authenticated user if it exists.
+     *
+     * @throws IllegalStateException when the authenticated user is not configured for the client.
+     */
+    public fun authenticatedUser(): UserId {
+        if (::user.isInitialized) {
+            return user
+        }
+        throw Exceptions.newIllegalStateException(
+            "User has not been authenticated in the system."
+        )
+    }
+
+    /**
+     * Returns the ID of the user's wallet if it exists.
+     *
+     * @throws IllegalStateException when the user's wallet does not exist
+     * because the authenticated user is not configured to the client.
+     */
+    public fun wallet(): WalletId {
+        if (::walletId.isInitialized) {
+            return walletId
+        }
+        throw Exceptions.newIllegalStateException(
+            "There is no user's wallet ID because " +
+                    "the user has not been authenticated in the system."
+        )
+    }
+
+    /**
+     * Returns the `CreateWallet` command.
+     *
+     * @param wallet the ID of the wallet to create
+     */
+    private fun createWallet(wallet: WalletId): CreateWallet {
```

Comment:
Let's make this one an extension function to `CreateWallet`.

Reply:
As far as I understand, it will be great to create an API like this `CreateWallet.withId(id)`. The problem is that the `CreateWallet` type like any other type from `Proto` files generated to Java and doesn't have a `Companion` object to add a static extension for. Of course, we can add an extension function for the `Builder` of the `CreateWallet` or add an extension for the `WalletId` type, but I think that is not what we want to do. 
There are some issues that I have found about static extensions: 
<a href="https://youtrack.jetbrains.com/issue/KT-11968">Issue on YouTrack</a>
<a href="https://github.com/Kotlin/KEEP/issues/348">Issue on GitHub</a>

Reply:
Let's then have an extension to `Builder`.

@Vladyslav-Kuksiuk FYI.

Reply:
@Artem-Semenov-dev @Vladyslav-Kuksiuk FYI.

I have tried this one myself, with the types I have in ProtoData. 

Here are the declarations I used:

```proto
message ProtobufSourceFile {
    option (entity).kind = VIEW;

    FilePath file_path = 1;

    // ...
}
```

```kotlin
private fun ProtobufSourceFile.takeMe(toChurch: String) : StringValue {
    return StringValue
        .newBuilder()
        .setValue(this.filePath.value + toChurch)
        .build()
}

public fun main() {
    val path = FilePath.newBuilder().setValue("/Earth/1973/Aerosmith/Dream")
    val protoFile = ProtobufSourceFile
        .newBuilder()
        .setFilePath(path)
        .build()

    val taken = protoFile.takeMe("On")
    println(taken.toJson())
}
```

And here is the output:
```
"/Earth/1973/Aerosmith/DreamOn"
```